### PR TITLE
cleanup stratum errors

### PR DESCRIPTION
**Pull request by debris**

### CODE REVIEW

**Comment by ordian** [comment]
What's the motivation behind replacing `Error`s with `String`s?

**Comment by debris** [comment]
Getting rid of bloated error types which contain the same kind of errors on multiple levels and are never matched. 

I believe that if the 'error' is created only in one place and is never matched it does not deserve it's own type

**Comment by dvdplm** [comment]
I'm not sure I agree with this either, in particular I think that a proper error type helps readability – at a glance I know what the anticipated error conditions for a crate/module are. New devs looking at the code are going to go look for the errors too, and if they have to chase around for strings sprinkled all over the place they'll hate us.

@debris when you say that the error types are bloated, what are you referring to? You mean actually bloated as in "a potential performance problem" or more "unneeded type wankery"?

File: miner/stratum/src/lib.rs [hunk 1]
```diff
@@ -204,13 +204,11 @@ impl StratumImpl {
...
 		if let Some(job) = self.dispatcher.job() {
-			if let Err(e) = self.push_work_all(job, tcp_dispatcher) {
-				warn!("Failed to update some of the peers: {:?}", e);
-			}
+			self.push_work_all(job, tcp_dispatcher)
 		}
 	}
 
-	fn push_work_all(&self, payload: String, tcp_dispatcher: &Dispatcher) -> Result<(), Error> {
+	fn push_work_all(&self, payload: String, tcp_dispatcher: &Dispatcher) {
```
[the remark this answers is not in the document]
Comment (dvdplm) [reply miner/stratum/src/lib.rs]: Addressed this in https://github.com/paritytech/parity-ethereum/pull/11161

**Comment by dvdplm** [comment]
Closing as stale.

**Pull request closed without merging**